### PR TITLE
[SPARK-47309][SQL][XML] Add schema inference unit tests

### PR DESCRIPTION
**Pull request by shujingyang-db**
### What changes were proposed in this pull request?

As titled.

### Why are the changes needed?

Fix a bug.

### Does this PR introduce _any_ user-facing change?

Yes

### How was this patch tested?

Unit tests

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlInferSchemaSuite.scala [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+          StructField("field2", ArrayType(StringType, true), true) :: Nil
+        ),
+        true
+      ) :: Nil
+    )
+    assert(expectedSchema === xmlDF.schema)
+  }
+
+  test("complex arrays") {
+    val xmlDF = readData(complexFieldAndType2)
```
Comment (sandip-db) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlInferSchemaSuite.scala:199]: Is the omission of `arrayOfStruct` and `complexArrayOfStruct` intentional?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala [hunk 2]
```diff
@@ -68,4 +68,444 @@ private[xml] trait TestXmlData {
...
+          |        <inner4>3</inner4>
+          |      </inner3>
+          |      <inner3/>
+          |    </item>
+          |  </array>
+          |</arrayOfArray2>
+          |</ROW>
+          |""".stripMargin :: Nil
+
+  def nullsInArrays: Seq[String] =
```
Comment (sandip-db) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala:412]: No test for this?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala [hunk 3]
```diff
@@ -68,4 +68,444 @@ private[xml] trait TestXmlData {
...
+              <array>
+                <item>1</item>
+                <item>2</item>
+                <item>3</item>
+              </array>
+            </array>
+          </field4>
+        </ROW>""" :: Nil
+
+  def corruptRecords: Seq[String] =
```
Comment (sandip-db) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala:465]: unused

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala [hunk 4]
```diff
@@ -68,4 +68,444 @@ private[xml] trait TestXmlData {
...
+    """<ROW>
+          <b>
+            <item>
+              <c><struct></struct></c>
+            </item>
+            <item/>
+          </b>
+        </ROW>""" :: Nil
+
+  def arrayAndStructRecords: Seq[String] =
```
Comment (sandip-db) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala:501]: unused

**Review by sandip-db — APPROVED** [review]

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlInferSchemaSuite.scala (hunk 1 above)
Comment (shujingyang-db) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlInferSchemaSuite.scala:199]: Yes. We converted the array of arrays into arrays of struct as XML doesn't support arrays of arrays. I will remove these for clarity.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala (hunk 2 above)
Comment (shujingyang-db) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala:412]: I plan to add tests in a follow-up PR. I will remove this for clarity in this PR.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala (hunk 3 above)
Comment (shujingyang-db) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala:465]: I plan to add tests in a follow-up PR. I will remove this for clarity in this PR.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala (hunk 4 above)
Comment (shujingyang-db) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/TestXmlData.scala:501]: I plan to add tests in a follow-up PR. I will remove this for clarity in this PR.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**